pkg/deploy/controller/deploymentconfig/deploymentconfig_controller.go

```diff
@@ -84,11 +89,29 @@ func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig)
 		return c.updateStatus(config, []*kapi.ReplicationController{})
 	}
 
-	// Find all deployments owned by the deployment config.
+	// List all ReplicationControllers to find also those we own but that no longer match our selector.
+	// They will be orphaned by ClaimReplicationControllers().
+	rcList, err := c.rcLister.ReplicationControllers(config.Namespace).List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("DeploymentConfigController can't list all RCs: %s", err)
+	}
 	selector := deployutil.ConfigSelector(config.Name)
-	existingDeployments, err := c.rcLister.ReplicationControllers(config.Namespace).List(selector)
+	// If any adoptions are attempted, we should first recheck for deletion with
+	// an uncached quorum read sometime after listing ReplicationControllers (see Kubernetes #42639).
+	canAdoptFunc := kcontroller.RecheckDeletionTimestamp(func() (metav1.Object, error) {
+		fresh, err := c.dn.DeploymentConfigs(config.Namespace).Get(config.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+		if fresh.UID != config.UID {
+			return nil, fmt.Errorf("original DeploymentConfig %v/%v is gone: got uid %v, wanted %v", config.Namespace, config.Name, fresh.UID, config.UID)
+		}
+		return fresh, nil
+	})
+	cm := oscontroller.NewRCControllerRefManager(c.rcControl, config, selector, deployutil.ControllerKind, canAdoptFunc)
+	existingDeployments, err := cm.ClaimReplicationControllers(rcList)
 	if err != nil {
-		return err
+		return fmt.Errorf("DeploymentConfigController failed to claim RCs: %s", err)
```

Comment:
%v (we use %v for errors in general, and %s for strings (sometimes %q for string but I forgot what the rule was ;-))